xija/runner.py

```diff
@@ -0,0 +1,212 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Efficiently run multiple xija models on a list of OR's.
+"""
+
+from chandra_aca.transform import calc_aca_from_targ
+
+
+DEFAULT_MODELS = ('aca', 'dpa', 'dea', 'acisfp', 'tcylaft6')  # etc
+
+
+class ModelComp:
+    """
+    Descriptor to specify a model component
+    """
+    def __init__(self, limit=None, model_err=None):
+        self.limit = limit
+        self.model_err = model_err
+
+    def __get__(self, instance, owner_cls):
+        if instance is None:
+            # When called without an instance, return self to allow access
+            # to descriptor attributes.
+            return self
+        else:
+            return getattr(instance, self.name)
+
+    def __set__(self, instance, value):
+        setattr(instance, self.name, value)
+
+    def __set_name__(self, owner_cls, name):
+        self.name = name
+
+        if not hasattr(owner_cls, 'comp_names'):
+            owner_cls.comp_names = []
+        owner_cls.comp_names.append(name)
+
+
+class ModelRunner:
+    """
+    Class object that knows how to run a particular model.  This
+    needs to be subclassed with specifics of the model components
+    that needs to be set and relevant model values that are output.
+    """
+    def run(self, obs_req, start):
+        """
+        Run the model for the specified ``obs_req``
+
+        This sets an attribute ``final_state``
+        """
+        self.states = self.make_states(obs_req)
+
+        # Now make the model, initialize the states, run, and set ``final_state``
+
+    def make_states(self, obs_req, start):
+        """
+        Make state values required to run the model for ``obs_req``.
+
+        Includes:
+
+        - Generate a maneuver to the new attitude
+        - Assume a SIM translation (if necessary) in the middle of the maneuver
+          (is there a better choice?)
+        - Potentially replicate some of the ORviewer logic for going to zero-FEPs.
+          If this is not done then the answers for ACIS models are conservative.
+        """
+        pass
+
+    def check_against_limits(self):
+        """
+        Check that the computed temperature(s) are within specified limits.
+
+        This could in theory be a tri-state
+        value: OK, not OK, and almost OK (meaning that the worst case violation
+        was within e.g. 20% of the model error).
```

Comment:
It might be better to return the difference from the model limits and let the end user application determine if that is OK, Not OK, etc.  For example, for a model that is well within the limits, the final application might show a green indicator, but for a model that is near one of the limits, (within some percentage as described in the stub comments here, the raw value is displayed for end user evaluation under a warning yellow/orange indicator.